### ISSUE TITLE
Add UI element to calculate the distance between two Landmarks

### ISSUE THREAD

**Issue by NickolausDS** [informatics-isi-edu/mesh-viewer#34]
It should be possible to select two landmarks and calculate the euclidian distance between them. It should be easy to pick two landmarks arbitrarily of where they are on the model for the calculation. The final result should also be represented in real world values (such as meters) not coordinate space (which are likely arbitrary).

The units that define space units may need to be added to the spec. There should be two values:

1. The unit of measurement (Millimeters, Feet, Miles, Kilometers)
2. The conversion from XTK coordinate space to the defined units (For example: 1.0 units in coordinate space equals 1.25 meters)

**Comment by NickolausDS** [comment]
May be a duplicate of https://github.com/informatics-isi-edu/mesh-viewer/issues/33

**Comment by NickolausDS** [comment]
I'm still working on the UI part of this one, but my goal is to have the initial working version out today or tomorrow. I wanted to reuse existing components to reduce code duplication and ensure a consistent UI, but following the old code is taking longer than I expected. 

As a small aside, It appears something like this feature existed at some point but was removed after pick functionality stopped working. I spent some time trying to bring it back to life (which was actually spread across a couple duplicated modules, including ('all.html', and 'm.testviewer.js'), but wasn't able to get it working. Due to the 'pick' bug this functionality ultimately won't be a viable path, so I've looked into driving it through the UI instead.

**Comment by NickolausDS** [comment]
Do the calculations by selecting the points you want to compare from the 'Meshes' panel then clicking 'calculate'. Or, calculate distance between everything by simply clicking the 'Show Landmarks' button beforehand. 

@robes The first pass looks like this, before I continue, do you have any feedback? 

<img width="1280" alt="screen shot 2018-06-11 at 1 44 38 pm" src="https://user-images.githubusercontent.com/865553/41256421-6487552a-6d7e-11e8-893f-befbc1a7ed5a.png">

**Comment by robes** [comment]
I don't think the UX would work well showing all pairwise calculations at once; not as the number of landmarks increase at least. I think we'd want something where you pick the pair of points and then show the distance (or calculate and show). You could share it with Bridget and Vu if you want their feedback.

**Comment by NickolausDS** [comment]
Currently if the number of landmarks is a problem, you can only select the ones you want to use from the 'meshes' panel, and only those will show up in the list.

**Comment by NickolausDS** [comment]
> I think we'd want something where you pick the pair of points and then show the distance (or calculate and show).

Actually, I think this is what the original functionality did (and is more ideal for UX), but since the 'pick' bug doesn't allow us to use click events in the mesh viewer, picking the meshes needs to be driven through the UI. 

Here's what computing only selected points looks like with current changes: 

<img width="1055" alt="screen shot 2018-06-11 at 4 05 11 pm" src="https://user-images.githubusercontent.com/865553/41261700-c506ecf4-6d91-11e8-9006-525f8674d6d0.png">
<img width="989" alt="screen shot 2018-06-11 at 4 05 27 pm" src="https://user-images.githubusercontent.com/865553/41261724-e6aedcb8-6d91-11e8-9fe8-8cdad27eebb7.png">

**Comment by robes** [comment]
the last issue is fixed. but there is a scrolling issue maybe in general with the mesh sidebar and issue with the text going beyond the boundary.

![image](https://user-images.githubusercontent.com/4202842/47742295-97090f00-dc39-11e8-9914-cdb2a4f41a82.png)

**Comment by robes** [comment]
Also is there a way now to specify the units? I see in your screenshot "40.93cm" but in mine "2.45units" for example.

**Comment by NickolausDS** [comment]
Yes, two new fields have been added to the specification for the model, `measurement` and `unitconversion` (The last one is for converting units to world space, for example if 1 unit of world space was 2.4 centimeters). Mesh viewer will now throw warnings if one or both are not set, and default to 'units' in world space if they're not.
